pkg/internalapis/cnsvolumeoperationrequest/cnsvolumeoperationrequest.go

```diff
@@ -120,22 +107,124 @@ func InitVolumeOperationRequestInterface(ctx context.Context) (VolumeOperationRe
 // Returns an error if any error is encountered while attempting to
 // read the previously persisted information from the API server.
 // If no previous information has been persisted, it returns an empty object.
-func (or *operationRequestStoreOnETCD) GetRequestDetails(ctx context.Context, name string) (*VolumeOperationRequestDetails, error) {
+func (or *operationRequestStore) GetRequestDetails(ctx context.Context, name string) (*VolumeOperationRequestDetails, error) {
 	log := logger.GetLogger(ctx)
-	// TODO: Implement GetRequestDetails for operationRequestStoreOnETCD
-	err := fmt.Errorf("GetRequestDetails not implemented for operationRequestStoreOnETCD")
-	log.Error(err)
-	return nil, err
+	namespacedName := client.ObjectKey{Name: name, Namespace: csiconfig.DefaultCSINamespace}
+	log.Infof("Getting CnsVolumeOperationRequest instance with name %s/%s", namespacedName.Namespace, namespacedName.Name)
+
+	instance := &cnsvolumeoperationrequestv1alpha1.CnsVolumeOperationRequest{}
+	err := or.k8sclient.Get(ctx, namespacedName, instance)
+	if err != nil {
+		// If instance does not exist, return an empty object.
+		if apierrors.IsNotFound(err) {
+			log.Infof("CnsVolumeOperationRequest instance %s/%s not found. Returning ...", namespacedName.Namespace, namespacedName.Name)
+			return nil, nil
```

Comment:
If an error is not returned and if the caller assumes the returned object is not nil, then we will have panic. It is better to return a not found error to the caller, rather than returning nil error with nil object.

Reply:
makes sense. Removed this part of the logic. Callers will handle the not found error